lib/pdf/collate.rb

```diff
@@ -44,13 +47,13 @@ def self.merge_pdfs(output_file, *filenames)
     temp_file_handles = []
 
     filenames.map! do |filename|
-      next filename unless URI.parse(filename).scheme == 'http'
+      next filename unless string_is_url(filename)
       begin
         local_file = Tempfile.from_url(filename)
       rescue Exception => msg
         puts "Error downloading PDF file #{filename} for output file #{output_file}. Aborting"
         puts "Error message: #{msg}"
-        exit 1
+        raise
```

Comment:
I see you are rescuing this in some places and logging to hipchat, however there is one place that you're not catching/logging: https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites/virtual/collate_pdfs.rake#L22

I'd propose either just logging to hipchat from here and then exiting, or else making sure you're catching / logging in collate_pdfs.rake